### DISCUSSION TITLE
How to ignore lowercase enum validation error?

### DISCUSSION THREAD

**Discussion by katerynaprotsenko** [ChilliCream/hotchocolate#5296] [Q&A] [answered]
Hello, we have an input object graph type for an object which contains a enum field. 
According to the documentation _"Enum values are automatically formatted to the UPPER_SNAIL_CASE according to the GraphQL specification"_
So, when we pass a lowercase enum value into IDE, validation fails. Is there any way to ignore it? We would like to change that value to uppercase in middleware, but due to the validation we never reach it.

**Comment by michaelstaib** [comment, marked as answer]
Hey there,

So at the moment that does not work since we do an exact match in the type. But I will open this up in the type a bit so that you can define your own equality comparer to enum types.

I think we will do another V12 version next week and we will include this there.

**Reply by katerynaprotsenko** [reply]
Thank you for the quick response!

**Reply by michaelstaib** [reply]
https://github.com/ChilliCream/hotchocolate/pull/5297

**Reply by michaelstaib** [reply]
So... 

here is a preview ... will take an hour to be available on NuGet:
https://github.com/ChilliCream/hotchocolate/releases/tag/12.13.0-preview.1

You can now specify the name equality comparer on the EnumType:

```csharp
public class FooType : EnumType<Foo> 
{
    protected override void Configure(IEnumTypeDescriptor<T> descriptor) 
    {
        descriptor.NameComparer(NameStringComparer.OrdinalIgnoreCase);
    }
}
```

If you want to apply that to all enum types that you have without specifying this explicitly on the enum type then you can use a type interceptor like the following:

```csharp
public sealed class EnumTypeInterceptor : TypeInterceptor
{
    public override void OnAfterRegisterDependencies(
        ITypeDiscoveryContext discoveryContext,
        DefinitionBase definition,
        IDictionary<string, object> contextData)
    {
        if (definition is EnumTypeDefinition enumTypeDef)
        {
            enumTypeDef.NameComparer = StringComparer.OrdinalIgnoreCase;
        }
    }
}
```

Register the type interceptor like the following:

```csharp
services
    .AddGraphQLServer()
    .TryAddTypeInterceptor<EnumTypeInterceptor>()
```

Have fun.

**Reply by michaelstaib** [reply]
In the meanwhile its now on NuGet:
https://www.nuget.org/packages/HotChocolate/12.13.0-preview.1

**Reply by katerynaprotsenko** [reply]
Wow! Thank you so much!
